src/main/java/duke/Command.java

```diff
@@ -0,0 +1,221 @@
+     * Execute the command to chat with the user.
+     *
+     * @param scanner  the scanner object to get user input
+     * @param allTasks the list to store all tasks created by the user
+     */
+    public static void chat(Scanner scanner, AllTasksList allTasks) {
+        while (isChatting) {
+            try {
+                System.out.println(BORDER);
+                System.out.print(": ");
+
+                String userInput = scanner.nextLine();
+                Command.parseAndExecuteCommand(userInput, scanner, allTasks);
+
+                System.out.println(BORDER + "\n");
+                Storage.storeTasks(allTasks);
+            } catch (DukeException e) {
+                System.out.println(e.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Execute the command to mark a task as complete.
+     *
+     * @param commandArray a string array of commands to be parsed for more information
+     * @param allTasks     the list to store all tasks created by the user
+     * @throws DukeException
+     */
+    public static void markTask(String[] commandArray, AllTasksList allTasks)
+            throws DukeException {
+        try {
+            if (commandArray.length <= 1) {
+                throw noNumberException;
+            }
+
+            int index = Integer.parseInt(commandArray[1]) - 1;
+            allTasks.markTask(index);
+        } catch (NumberFormatException e) {
+            throw noNumberException;
+        }
+    }
+
+    /**
+     * Execute the command to mark a task as incomplete.
+     *
+     * @param commandArray a string array of commands to be parsed for more information
+     * @param allTasks     the list to store all tasks created by the user
+     * @throws DukeException
+     */
+    public static void unMarkTask(String[] commandArray, AllTasksList allTasks)
+            throws DukeException {
+        try {
+            if (commandArray.length <= 1) {
+                throw noNumberException;
+            }
+
+            int index = Integer.parseInt(commandArray[1]) - 1;
+            allTasks.unMarkTask(index);
+        } catch (NumberFormatException e) {
+            throw noNumberException;
+        }
+    }
+
+    /**
+     * Execute the command to delete a task.
+     *
+     * @param commandArray a string array of commands to be parsed for more information
+     * @param allTasks     the list to store all tasks created by the user
+     * @throws DukeException
+     */
+    public static void delete(String[] commandArray, AllTasksList allTasks)
+            throws DukeException {
+        try {
+            if (commandArray.length <= 1) {
+                throw noNumberException;
+            }
+
+            int index = Integer.parseInt(commandArray[1]) - 1;
+            allTasks.delete(index);
+        } catch (NumberFormatException e) {
+            throw noNumberException;
+        }
+    }
+
+    /**
+     * utility method used to parse the and execute the user command.
```

Comment:
I think you missed out a capital U in `utility` here. No other comments so I'm grasping for straws here >-<